guides/configuration.md

```diff
@@ -0,0 +1,139 @@
+## Custom service classes
+
+The Solidus configuration object from above has some getters and setters for classes as extension points. After evaluating whether you need to go down this route, take some time to study the standard implementation for the class you want to modify, and subclass or re-implement it. The service class extension points can be found [here](../core/app/models/spree/app_configuration.rb#L277-L383).
+
+For example, you might want the Product index page to behave differently than by default. The [default search class](../core/lib/spree/core/search/base.rb) defines a private `#get_base_scope` method which we can override in a class that inherits from that default search class.
+
+Build your own searcher in `lib/mystore/product_search.rb`:
+
+```
+module MyStore
+  class ProductSearch < Spree::Core::Search::Base
+    private
+    def get_base_scope
+      super.where("name LIKE '%rails%'")
+    end
+  end
+end
+```
+
+What this new searcher does is restricting the search results to only show products with the string "rails" in their name. Neat!
+
+In order to integrate this searcher in your app, add the following to your Solidus initializer in `config/initializers/spree.rb`:
+
+```
+require 'my_store/product_search'
+
+Spree.config do |config|
+  config.use_static_preferences!
+
+  # code ommitted here
+
+  config.searcher_class = MyStore::ProductSearch
+end
+```
+
+Your search will now only return products that contain "rails" in their name.
+
+This pattern is our preferred pattern for generating extension points. If you see yourself overriding a class with a definable interface, and there is no such extension point, we're very happy to accept a Pull Request implementing that changeable class.
+
+## Monkey-Patching Solidus
+
+Solidus is setup so that any file in the `app` directory with the suffix `_decorator.rb` will be auto-loaded like any model or controller files.
+
+For example, if you want to add a method to the `Spree::Order` model, you would have a file called `app/models/mystore/order_decorator.rb` with the following contents:
+
+```
+module MyStore
+  module OrderDecorator
+    def total
+      super + BigDecimal.new(10.0)
+    end
+  end
+end
+```
+
+In order to override the `total` method on `Spree::Order`, add the following to the file `app/models/spree/order_decorator.rb`:
+
+```
+Spree::Order.prepend MyStore::OrderDecorator
+```
+
+This creates a new module called `MyStore::OrderDecorator` and will insert its methods so early in the method lookup chain for method calls on `Spree::Order` objects that the decorator's methods override the original methods.
+
+From now on, every order, when asked for its total, will return the an inflated total by 10 Dollars (or whatever your currency is).
+
+If you do this kind of thing, be very careful and test your new functionality very well. If you do upgrades, read extra carefully about changes in Solidus' core classes.
+
+## Run a private fork (and submit us a Pull Request!)
```

Comment:
Could you please remove the call for pull requests? This is a little bit too "loud" ;)